job-server/src/spark.jobserver/JobServerNamedObjects.scala

```diff
@@ -96,5 +93,15 @@ class JobServerNamedObjects(system: ActorSystem) extends NamedObjects {
       case answer: Iterable[String] @unchecked => answer
     }
   }
+}
 
+/**
+ * companion object that hold reference to cache with named object so that we can reference
+ * named objects across jobs
+ */
+object JobServerNamedObjects {
+  // we must store a reference to each NamedObject even though only its ID is used here
+  // this reference prevents the object from being GCed and cleaned by sparks ContextCleaner
+  // or some other GC for other types of objects
+  val namesToObjects: Cache[NamedObject] = LruCache()
```

Comment:
This would work for contexts that are forked, but would not work when `context-per-ivm` is false.